debugger/src/rtt.rs

```diff
@@ -0,0 +1,388 @@
+#[derive(Debug)]
+pub struct RttActiveChannel {
+    pub up_channel: Option<UpChannel>,
+    pub down_channel: Option<DownChannel>,
+    pub channel_name: String,
+    pub data_format: DataFormat,
+    /// Data that will be written to the down_channel (host to target)
+    input_data: String,
+    rtt_buffer: RttBuffer,
+    show_timestamps: bool,
+}
+
+/// A fully configured RttActiveChannel. The configuration will always try to 'default' based on information read from the RTT control block in the binary. Where insufficient information is available, it will use the supplied configuration, with final hardcoded defaults where no other information was available.
+impl RttActiveChannel {
+    pub fn new(
+        up_channel: Option<UpChannel>,
+        down_channel: Option<DownChannel>,
+        channel_config: Option<RttChannelConfig>,
+    ) -> Self {
+        let full_config = match channel_config {
+            Some(channel_config) => channel_config,
+            None => RttChannelConfig {
+                ..Default::default() // Will set intelligent defaults below ...
+            },
+        };
+        let name = up_channel
+            .as_ref()
+            .and_then(|up| up.name().map(Into::into))
+            .or_else(|| {
+                down_channel
+                    .as_ref()
+                    .and_then(|down| down.name().map(Into::into))
+            })
+            .or_else(|| full_config.clone().channel_name)
+            .unwrap_or_else(|| "Unnamed RTT channel".to_string());
+        let buffer_size: usize = up_channel
+            .as_ref()
+            .map(|up| up.buffer_size())
+            .or_else(|| down_channel.as_ref().map(|down| down.buffer_size()))
+            .unwrap_or(1024); // This should never be the case ...
+        let defmt_enabled: bool = up_channel
+            .as_ref()
+            .map(|up| up.name() == Some("defmt"))
+            .or_else(|| {
+                down_channel
+                    .as_ref()
+                    .map(|down| down.name() == Some("defmt"))
+            })
+            .unwrap_or(false); // This should never be the case ...
+        let data_format: DataFormat = if defmt_enabled {
+            DataFormat::Defmt
+        } else {
+            full_config.data_format
+        };
+        Self {
+            up_channel,
+            down_channel,
+            channel_name: name,
+            data_format,
+            input_data: String::new(),
+            rtt_buffer: RttBuffer::new(buffer_size),
+            show_timestamps: full_config.show_timestamps,
+        }
+    }
+
+    /// Returns the number of the UpChannel
+    pub fn number(&self) -> Option<usize> {
+        self.up_channel.as_ref().map(|uc| uc.number())
+    }
+
+    /// Polls the RTT target for new data on the specified channel.
+    /// Processes all the new data into the channel `rtt_buffer` and returns the number of bytes that was read
```

Comment:
```suggestion
    /// Polls the RTT target for new data on the channel represented by `self`.
    /// Processes all the new data into the channel internal buffer and returns the number of bytes that was read.
```

`rtt_buffer` is a private member and should not be mentioned in the public doc comment imo. Also, I tried to make the explanation more clear :)